### ISSUE TITLE
Unable to determine dependencies for Winter CMS. Invalid version string "v1.2.x-dev abbdbb7"

### ISSUE THREAD

**Issue by dynamiccontent** [wintercms/web-installer#31]
Debian GNU/Linux 11 (bullseye)
PHP 8.2.0RC5
10.9.3-MariaDB-1:10.9.3+maria~deb11 mariadb.org binary distribution
nginx version: nginx/1.23.2
php-fpm 8.2

install.log

```
[2022-11-04 18:18:33+00:00] INFO: Installer API request received {"method":"GET","endpoint":"checkApi"}
[2022-11-04 18:18:33+00:00] NOTICE: Trying Winter CMS API
[2022-11-04 18:18:33+00:00] INFO: Winter API request {"method":"GET","uri":"ping"}
[2022-11-04 18:18:33+00:00] INFO: Installer API request received {"method":"GET","endpoint":"checkPhpVersion"}
[2022-11-04 18:18:33+00:00] NOTICE: Compared PHP version {"installed":"8.2.0RC5","needed":"8.0.2"}
[2022-11-04 18:18:33+00:00] NOTICE: PHP version requirement met.
[2022-11-04 18:18:33+00:00] INFO: Installer API request received {"method":"GET","endpoint":"checkPhpExtensions"}
[2022-11-04 18:18:33+00:00] NOTICE: Checking PHP "curl" extension
[2022-11-04 18:18:33+00:00] NOTICE: Checking PHP "json" extension
[2022-11-04 18:18:33+00:00] NOTICE: Checking PHP "pdo" extension
[2022-11-04 18:18:33+00:00] NOTICE: Checking PHP "zip" extension
[2022-11-04 18:18:33+00:00] NOTICE: Checking PHP "mbstring" extension
[2022-11-04 18:18:33+00:00] NOTICE: Checking PHP "fileinfo" extension
[2022-11-04 18:18:33+00:00] NOTICE: Checking PHP "openssl" extension
[2022-11-04 18:18:33+00:00] NOTICE: Checking PHP "gd" extension
[2022-11-04 18:18:33+00:00] NOTICE: Checking PHP "filter" extension
[2022-11-04 18:18:33+00:00] NOTICE: Checking PHP "hash" extension
[2022-11-04 18:18:33+00:00] NOTICE: Checking PHP "dom" extension
[2022-11-04 18:18:33+00:00] NOTICE: Required PHP extensions are installed.
[2022-11-04 18:18:34+00:00] NOTICE: Response received from Winter CMS API {"response":"pong"}
[2022-11-04 18:18:34+00:00] NOTICE: Winter CMS API connection successful.
[2022-11-04 18:21:19+00:00] INFO: Installer API request received {"method":"POST","endpoint":"checkDatabase"}
[2022-11-04 18:21:19+00:00] NOTICE: Check database connection
[2022-11-04 18:21:19+00:00] INFO: Installer API request received {"method":"GET","endpoint":"checkWriteAccess"}
[2022-11-04 18:21:19+00:00] NOTICE: Current working directory is writable.
[2022-11-04 18:21:19+00:00] NOTICE: Found 0 table(s) {"tables":""}
[2022-11-04 18:21:19+00:00] NOTICE: Database connection established and verified empty
[2022-11-04 18:21:23+00:00] INFO: Installer API request received {"method":"POST","endpoint":"downloadWinter"}
[2022-11-04 18:21:23+00:00] NOTICE: Try downloading Winter CMS archive
[2022-11-04 18:21:23+00:00] NOTICE: Downloading Winter ZIP via cURL {"url":"https://github.com/wintercms/winter/archive/refs/heads/1.2.zip"}
[2022-11-04 18:21:26+00:00] NOTICE: Winter CMS ZIP file downloaded {"path":"/var/www/winter/.wintercms/winter.zip"}
[2022-11-04 18:21:26+00:00] INFO: Installer API request received {"method":"POST","endpoint":"extractWinter"}
[2022-11-04 18:21:26+00:00] NOTICE: Begin extracting Winter CMS archive
[2022-11-04 18:21:27+00:00] NOTICE: Move subfoldered files into position {"subfolder":"winter-1.2/"}
[2022-11-04 18:21:27+00:00] NOTICE: Remove ZIP subfolder {"subfolder":"winter-1.2/"}
[2022-11-04 18:21:27+00:00] NOTICE: Make artisan command-line tool executable {"path":"/var/www/winter/.wintercms/artisan"}
[2022-11-04 18:21:27+00:00] NOTICE: Remove Composer Merge plugin from composer.json if found
[2022-11-04 18:21:27+00:00] NOTICE: Found merge plugin in required packages - removing
[2022-11-04 18:21:27+00:00] NOTICE: Found merge plugin config in "extra" config definition - removing
[2022-11-04 18:21:27+00:00] INFO: Installer API request received {"method":"POST","endpoint":"lockDependencies"}
[2022-11-04 18:21:27+00:00] NOTICE: Create Composer instance
[2022-11-04 18:21:27+00:00] NOTICE: Set memory limit to 1.5GB
[2022-11-04 18:21:27+00:00] NOTICE: Set work directory for Composer {"path":"/var/www/winter/.wintercms"}
[2022-11-04 18:21:27+00:00] NOTICE: Create home/cache directory for Composer {"path":"/var/www/winter/.wintercms/.composer"}
[2022-11-04 18:21:27+00:00] NOTICE: Set home/cache directory for Composer {"path":"/var/www/winter/.wintercms/.composer"}
[2022-11-04 18:21:27+00:00] NOTICE: Run Composer "update" command - generate only a lockfile
[2022-11-04 18:21:30+00:00] ERROR: Unable to determine dependencies for Winter CMS. Invalid version string "v1.2.x-dev abbdbb7" {"code":500,"exception":null}

```
![wintercms](https://user-images.githubusercontent.com/56940608/200050059-e4507bac-d79f-4fb4-bd39-260815c67297.png)

**Comment by LukeTowers** [comment]
@dynamiccontent we haven't run any tests on PHP 8.2, try switching to 8.1 instead?

**Comment by merodeador** [comment]
I've just had this error. Both the web installer and the install.log look the same but the 'invalid' version is "6.3.x-dev 2e2ac75".
The installed PHP version is 8.1.10, mySQL 8.0.30 on a windows 10 machine running Laragon.

Installing via composer worked just fine.

**Comment by bennothommo** [comment]
@merodeador if you try the Web Installer again and hit this same issue, could you look in the folder that you ran the installation from - there should be a hidden `.wintercms` folder that contains the partial install. Would you be able to copy the contents of `composer.json` into here for us?

**Comment by merodeador** [comment]
Sure, this is it:
`{
    "name": "wintercms/winter",
    "description": "Free, open-source, self-hosted CMS platform based on the Laravel PHP Framework. Originally known as October CMS.",
    "homepage": "https://wintercms.com",
    "type": "project",
    "keywords": [
        "winter",
        "cms",
        "wintercms",
        "laravel",
        "cmf"
    ],
    "license": "MIT",
    "authors": [
        {
            "name": "Alexey Bobkov",
            "email": "aleksey.bobkov@gmail.com",
            "role": "Original Author"
        },
        {
            "name": "Samuel Georges",
            "email": "daftspunky@gmail.com",
            "role": "Original Author"
        },
        {
            "name": "Luke Towers",
            "email": "wintercms@luketowers.ca",
            "role": "Lead Maintainer"
        }
    ],
    "support": {
        "issues": "https://github.com/wintercms/winter/issues",
        "docs": "https://wintercms.github.io/docs/",
        "discord": "https://discord.gg/D5MFSPH6Ux",
        "source": "https://github.com/wintercms/winter"
    },
    "require": {
        "php": "^8.0.2",
        "winter/storm": "~1.2.0",
        "winter/wn-system-module": "~1.2.0",
        "winter/wn-backend-module": "~1.2.0",
        "winter/wn-cms-module": "~1.2.0",
        "laravel/framework": "^9.1"
    },
    "require-dev": {
        "phpunit/phpunit": "^9.5.8",
        "mockery/mockery": "^1.4.4",
        "fakerphp/faker": "^1.9.2",
        "squizlabs/php_codesniffer": "^3.2",
        "php-parallel-lint/php-parallel-lint": "^1.0",
        "dms/phpunit-arraysubset-asserts": "^0.1.0|^0.2.1"
    },
    "scripts": {
        "post-create-project-cmd": [
            "@php artisan key:generate"
        ],
        "post-update-cmd": [
            "@php artisan winter:version",
            "@php artisan package:discover"
        ],
        "test": [
            "phpunit --stop-on-failure"
        ],
        "lint": [
            "parallel-lint --exclude vendor --exclude storage --exclude modules/system/tests/fixtures/plugins/testvendor/goto/Plugin.php ."
        ],
        "sniff": [
            "phpcs --colors -nq --report=\"full\" --extensions=\"php\""
        ]
    },
    "minimum-stability": "dev",
    "prefer-stable": true,
    "extra": [],
    "config": {
        "allow-plugins": {
            "composer/installers": true,
            "wikimedia/composer-merge-plugin": true
        }
    }
}`

**Comment by merodeador** [comment]
Sure, here it is:
```
{
    "name": "wintercms/winter",
    "description": "Free, open-source, self-hosted CMS platform based on the Laravel PHP Framework. Originally known as October CMS.",
    "homepage": "https://wintercms.com",
    "type": "project",
    "keywords": [
        "winter",
        "cms",
        "wintercms",
        "laravel",
        "cmf"
    ],
    "license": "MIT",
    "authors": [
        {
            "name": "Alexey Bobkov",
            "email": "aleksey.bobkov@gmail.com",
            "role": "Original Author"
        },
        {
            "name": "Samuel Georges",
            "email": "daftspunky@gmail.com",
            "role": "Original Author"
        },
        {
            "name": "Luke Towers",
            "email": "wintercms@luketowers.ca",
            "role": "Lead Maintainer"
        }
    ],
    "support": {
        "issues": "https://github.com/wintercms/winter/issues",
        "docs": "https://wintercms.github.io/docs/",
        "discord": "https://discord.gg/D5MFSPH6Ux",
        "source": "https://github.com/wintercms/winter"
    },
    "require": {
        "php": "^8.0.2",
        "winter/storm": "~1.2.0",
        "winter/wn-system-module": "~1.2.0",
        "winter/wn-backend-module": "~1.2.0",
        "winter/wn-cms-module": "~1.2.0",
        "laravel/framework": "^9.1"
    },
    "require-dev": {
        "phpunit/phpunit": "^9.5.8",
        "mockery/mockery": "^1.4.4",
        "fakerphp/faker": "^1.9.2",
        "squizlabs/php_codesniffer": "^3.2",
        "php-parallel-lint/php-parallel-lint": "^1.0",
        "dms/phpunit-arraysubset-asserts": "^0.1.0|^0.2.1"
    },
    "scripts": {
        "post-create-project-cmd": [
            "@php artisan key:generate"
        ],
        "post-update-cmd": [
            "@php artisan winter:version",
            "@php artisan package:discover"
        ],
        "test": [
            "phpunit --stop-on-failure"
        ],
        "lint": [
            "parallel-lint --exclude vendor --exclude storage --exclude modules/system/tests/fixtures/plugins/testvendor/goto/Plugin.php ."
        ],
        "sniff": [
            "phpcs --colors -nq --report=\"full\" --extensions=\"php\""
        ]
    },
    "minimum-stability": "dev",
    "prefer-stable": true,
    "extra": [],
    "config": {
        "allow-plugins": {
            "composer/installers": true,
            "wikimedia/composer-merge-plugin": true
        }
    }
}

```

**Comment by bennothommo** [comment]
Thanks everyone - I've been able to replicate this. Will have a fix out soon.

**Comment by bennothommo** [comment]
Hi all, I've released v1.2.2 of our Web Installer, which now contains a fix for this issue. Here's the direct link to the fixed version:
https://github.com/wintercms/web-installer/releases/download/v1.2.2/install.zip

**Comment by merodeador** [comment]
Just tried this, works perfectly. Thanks!
